plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,273 @@
+    >>> from astropy import units as u
+    >>> from plasmapy.dispersion import two_fluid_dispersion_solution as tfds
+    >>> k = 0.01 * u.m ** -1
+    >>> theta = 30 * u.deg
+    >>> B = 8.3E-9 * u.T
+    >>> n = 5.e6 * u.m ** -3
+    >>> T_e = 1.6e6 * u.K
+    >>> T_i = 4.e5 * u.K
+    >>> z = 1
+    >>> omega = tfds(B=B, k=k, n=n, T_e=T_e, T_i=T_i, theta=theta, z=z)
+    >>> omega
+    {'fast_mode': <Quantity [[1520.5794506]] rad / s>,
+     'alfven_mode': <Quantity [[1261.75471561]] rad / s>,
+     'acoustic_mode': <Quantity [[0.6881521]] rad / s>}
+
+    >>> k_arr = np.linspace(10**-7, 10**-2, 10000) * u.m ** -1
+    >>> theta = np.linspace(5, 85, 100) * u.deg
+    >>> n = 5.e6 * u.m ** -3
+    >>> B = 8.3E-9 * u.T
+    >>> T_e = 1.6e6 * u.K
+    >>> T_i = 4.e5 * u.K
+    >>> z = 1
+    >>> c = 3.e8 * u.m/u.s
+    >>> c_s = pfp.ion_sound_speed(T_e=T_e, T_i=T_i, n_e=z * n, ion='p+')
+    >>> v_A = pfp.Alfven_speed( B, n, ion='p+')
+    >>> omega_ci = pfp.gyrofrequency(B=B, particle='p+', signed=False, Z=z)
+    >>> omega = tfds(n=n, B=B, T_e=T_e, T_i=T_i, theta=theta, z=z, k=k_arr)
+    >>> omega['fast_mode'][:,40]
+    <Quantity [1.61176312e-02, 1.77335334e-01, 3.38688590e-01, ...,
+               1.52030361e+03, 1.52045553e+03, 1.52060745e+03] rad / s>
+    """
+
+    # Required derived parameters
+    # Compute the ion sound speed using the function from
+    # plasmapy.formulary.parameters
+    c_s = pfp.ion_sound_speed(
+        T_e=T_e, T_i=T_i, n_e=z * n, gamma_e=gamma_e, gamma_i=gamma_i, ion=ion
+    )
+
+    # Compute the ion Alfven speed using the function from
+    # plasmapy.formulary.parameters
+    v_A = pfp.Alfven_speed(B, n, ion=ion)
+
+    # Compute the ion gyro frequency using the function from
+    # plasmapy.formulary.parameters
+    omega_ci = pfp.gyrofrequency(B=B, particle="p+", signed=False, Z=z)
+
+    # Compute the electron plasma frequency using the function from
+    # plasmapy.formulary.parameters
+    omega_pe = pfp.plasma_frequency(n=n, particle="e-", z_mean=z)
+
+    # Compute the dimensionless parameters corresponding to equation 32 of
+    # Bellan2012JGR
+    alpha = (np.cos(theta.to("rad")) ** 2).value
+    beta = (c_s ** 2 / v_A ** 2).value
+
+    # Create a meshgrid of direction of propagation and the wavenumber
+    alphav, kv = np.meshgrid(alpha, k)
+
+    Lambda = (kv ** 2 * v_A ** 2 / omega_ci ** 2).value
+
+    # Compute the dimensionless parameters corresponding to equation 2 of
+    # Bellan2012JGR
+    Q = 1 + (kv ** 2 * c ** 2 / omega_pe ** 2).value
+
+    # Compute the dimensionless parameters corresponding to equation 35 of
+    # Bellan2012JGR
+    A = (Q + Q ** 2 * beta + Q * alphav + alphav * Lambda) / Q ** 2
+    B = alphav * (1 + 2 * Q * beta + Lambda * beta) / Q ** 2
+    C = alphav ** 2 * beta / Q ** 2
+
+    # Compute the dimensionless parameters corresponding to equation 36 of
+    # Bellan2012JGR
+    p = (3 * B - A ** 2) / 3
+    q = (9 * A * B - 2 * A ** 3 - 27 * C) / 27
```

Comment:
```suggestion
    # Bellan2012JGR params equation 32
    alpha = np.cos(theta.value) ** 2
    beta = (c_s / v_A).to(u.dimensionless_unscaled).value ** 2
    alphav, kv = np.meshgrid(alpha, k.value)  # create grid
    Lambda = (kv * v_A.value / omega_ci.value) ** 2

    # Bellan2012JGR params equation 2
    Q = 1 + (kv * c.value / omega_pe.value) ** 2

    # Bellan2012JGR params equation 35
    A = ((1 + alphav) / Q) + beta + (alphav * Lambda / Q ** 2)
    B = alphav * (1 + 2 * Q * beta + Lambda * beta) / Q ** 2
    C = beta * (alphav / Q) ** 2

    # Bellan2012JGR params equation 36
    p = (3 * B - A ** 2) / 3
    q = (9 * A * B - 2 * A ** 3 - 27 * C) / 27
```

Like above, this does not change the basic functionality but does...

1. reduce comments to improve readability
2. augments the order of operations to reduce the number of operations and avoid squaring large or small number before dividing
3. extracting the astropy `Quantity` values earlier to avoid issues around unit wrangling